src/imgui/imgui_raii_wrapper.h

```diff
@@ -0,0 +1,162 @@
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class TabBar {
+public:
+  [[nodiscard]] explicit TabBar(std::string const& id) : selected{ImGui::BeginTabBar(id.c_str())} {}
+  TabBar(TabBar const&) = delete;
+  TabBar& operator=(TabBar const&) = delete;
+  TabBar(TabBar&&) = delete;
+  TabBar& operator=(TabBar&&) = delete;
+  ~TabBar() {
+    if (selected) {
+      ImGui::EndTabBar();
+    }
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class TabItem {
+public:
+  [[nodiscard]] explicit TabItem(std::string const& id) : selected{ImGui::BeginTabItem(id.c_str())} {}
+  TabItem(TabItem const&) = delete;
+  TabItem& operator=(TabItem const&) = delete;
+  TabItem(TabItem&&) = delete;
+  TabItem& operator=(TabItem&&) = delete;
+  ~TabItem() {
+    if (selected) {
+      ImGui::EndTabItem();
+    }
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  bool selected{};
+};
+
+class Text {
+public:
+  Text(char const* text_) : text{text_} {}
+  Text(std::string text_) : text{std::move(text_)} {}
+  Text(char const* text_, ImVec4 color_) : text{text_}, color{color_} {}
+  Text(std::string text_, ImVec4 color_) : text{std::move(text_)}, color{color_} {}
+  std::string get_text() const {
+    return text;
+  }
+  std::optional<ImVec4> get_color() const {
+    return color;
+  }
+  void write() const {
+    if (color) {
+      ImGui::TextColored(color.value(), "%s", text.c_str());
+    } else {
+      ImGui::Text("%s", text.c_str());
+    }
+  }
+private:
+  std::string text{};
+  std::optional<ImVec4> color{};
+};
+
+class Table {
+public:
+  using Row = std::vector<Text>;
+  class Builder {
+  public:
+    Builder& with_id(std::string name) {
+      id = std::move(name);
+      return *this;
+    }
+    Builder& with_number_of_columns(std::size_t n) {
+      number_of_columns = n;
+      return *this;
+    }
+    Builder& with_headers(Row row) {
+      headers = std::move(row);
+      return *this;
+    }
+    Builder& add_row(Row row) {
+      values.push_back(std::move(row));
+      return *this;
+    }
+    Table build() {
+      return Table{
+        id,
+        static_cast<int>(number_of_columns),
+        headers,
+        values
+      };
+    }
+  private:
+    std::string id{};
+    std::size_t number_of_columns{};
+    Row headers{};
+    std::vector<Row> values{};
+  };
+
+  explicit Table(std::string const& id, int number_of_columns, Row const& headers, std::vector<Row> const& values) : selected{ImGui::BeginTable(id.c_str(), number_of_columns)} {
+    create_headers(headers);
+    for (auto const& row : values) {
+      create_row(row);
+    }
+  }
+  Table(Table const&) = delete;
+  Table& operator=(Table const&) = delete;
+  Table(Table&&) = delete;
+  Table& operator=(Table&&) = delete;
+  ~Table() {
+    ImGui::EndTable();
+  }
+  explicit operator bool() & {
+    return selected;
+  }
+private:
+  static void create_headers(Row const& labels) {
```

Comment:
What is static and how to use it? Why use const after row?

Reply:
https://mariusbancila.ro/blog/2018/11/23/join-the-east-const-revolution/ west vs east const

Reply:
as Luiz stated above there are many usages for the `const` keyword. 
Here we are passing a `const` reference of Rows into the function. 
You can think of a reference being the object with another name.
It's not a pointer nor a copy but the object itself (with another name) so we use it in situations like this, to evade the cost of a copy, but since we don't want to alter it, and we want to be sure we don't do it, we pass it as a `const`(ant) reference(`&`).
It's a very basic and common idiom in C++.